### PR TITLE
Rules which requires type information

### PR DESCRIPTION
**Pull request by unlight**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,8 +42,14 @@
     "typescript-eslint-parser": "10.0.0"
   },
   "lint-staged": {
-    "src/**/*": ["prettier-eslint --write", "git add"],
-    "test/**/*": ["prettier-eslint --write", "git add"]
```
Comment (unlight) [inline package.json]: Formatted by prettier

**Comment by JamesHenry** [comment]
Thank you, @unlight!

**Comment by JamesHenry** [comment]
The only thing I want to take a look into is the file resolution stuff, I will take a look this week and get back to you on here.

Thanks again!

**Comment by unlight** [comment]
Do not merge. Found a bug, when eslint working in watch mode, since program is a single for all files, it will not update source code when file is changed.

**Comment by unlight** [comment]
Fixed issue, I'll test it more on my simple ts project in this weekend.
Also, ts service was taken from my similar wrapper plugin https://github.com/unlight/eslint-plugin-tsc

**Comment by unlight** [comment]
@JamesHenry Are you going to merge this?
This stuff still has an issue, when eslint is working in watch mode (gulp-eslint)
I have no idea how to solve this.

**Pull request closed without merging**